Code/Framework/AtomCore/AtomCore/Instance/InstanceDatabase.h

```diff
@@ -391,41 +383,60 @@ namespace AZ
         Data::Instance<Type> InstanceDatabase<Type>::EmplaceInstance(
             const InstanceId& id, const Data::Asset<AssetData>& asset, const AZStd::any* param)
         {
-            // This assert is here to catch any potential non-randomness in our id generation. If it triggers,
-            // there might be a bug / race condition in the id generator. The same assert also occurs *after*
-            // instance creation to help differentiate between a non-random id vs recursive creation of the same id.
-            AZ_Assert(
-                !m_database.contains(id),
-                "Database already contains an instance for this id (%s), possibly a random id generation collision?",
-                id.ToString<AZStd::fixed_string<64>>().c_str());
-
-            // Emplace a new instance and return it.
+            // It's very important to have m_databaseMutex unlocked while an instance is being created because
+            // there can be cases like in StreamingImage(s), where multiple threads are involved and some of those threads
+            // attempt to release a StreamingImage, which in turn will lock m_databaseMutex and it could incurr
+            // in potential deadlocks.
+
+            // If the instance was created redundantly, it will be temporarily stored here for destruction
+            // before this function returns.
+            Data::Instance<Type> redundantInstance = nullptr;
+
             // It's possible for the m_createFunction call to recursively trigger another FindOrCreate call, so be aware that
             // the contents of m_database may change within this call.
             Data::Instance<Type> instance = nullptr;
-            if (!param)
+
             {
-                instance = m_instanceHandler.m_createFunction(asset.Get());
+                AZStd::scoped_lock<decltype(m_newInstanceMutex)> lock(m_newInstanceMutex);
```

Comment:
So now that releasing the instance and creating the instance use different mutexes the deadlock related to streaming images is fixed?

Reply:
Correct. This was the key to solve the deadlock related with StreamingImage Async Upload.
The reason it was not deadlocking on Vulkan was due to sheer luck because there was an additional Ref Count on the StreamingImage used by `ImageBasedLightComponentController`. I think the extra ref count was coming from the BindlessSrg. In Vulkan, A StreamingImage was not being released during `Secondary Copy Queue` and the deadlock wouldn't occur.